CompilerSource/gcc_interface/gcc_backend.cpp

```diff
@@ -132,33 +132,33 @@ const char* establish_bearings(const char *compiler)
 
   std::string MAKE_paths = compilerInfo.make_vars["PATH"];
   
-  std::string dirs = "CODEGEN=" + codegen_directory + " ";
-  dirs += "WORKDIR=" + eobjs_directory + " ";
+  std::string dirs = "CODEGEN=" + codegen_directory.string() + " ";
```

Comment:
If we're using `u8string()` above, why use `string()` elsewhere? How is `path.string().c_str()` wrong when `str + path.string()` is right?

Reply:
```path.c_str()``` was the one that was wrong. I missed this because it's not a c_str()